### PR TITLE
[v1.2-branch][ESP32][cherry-pick]Cherry pick some necessary fixes for v1.2-branch

### PR DESCRIPTION
**Pull request by Jerry-ESP**
Cherry pick some necessary fixes for v1.2-branch, include:
1. [1.1] Cherry pick Thread DNS client and memory leak fixes #31457
2. ESP32: Add EndpointQueueFilter for ESP32 platform #31440
3. [ESP32] Limit number of returned WiFi scan results to configured limit #30780
4. [ESP32] Fix the threading issue in nimble #29180
5. IM: Create ReadHandler after Session Establishment for Subscription Resumption #30491
6. [ESP32] Fix adding NDEBUG flag to CPPFLAGS #30763
7. Add records of session establishment for subscription resumption #31755
8. ESP32: check ap info in IsStationConnected #31438
9. Add checks for mOTInst in GenericThreadStackManagerImpl_OpenThread #32482
10. [ESP32] Fix few attributes with fixed quality in DeviceInfoProvider #32893
11. [ESP32] Implement BLE Manager Shutdown for nimble host #33109
12. [ESP32] Made a provision to generate esp_secure_cert partition in factory partition script. #29840
13. [ESP32] Added the support for onboarding paylaod in factory script #31274

### CODE REVIEW

**Comment by andy31415** [comment]
@Jerry-ESP - code lints are failing

**Comment by Jerry-ESP** [comment]
> @Jerry-ESP - code lints are failing

Updated

**Comment by Jerry-ESP** [comment]
> @Jerry-ESP - code lints are failing

Update

**Comment by Jerry-ESP** [comment]
> @Jerry-ESP - code lints are failing

Updated

**Review by andy31415 — APPROVED** [review]

File: scripts/tests/cirque_tests.sh [hunk 1]
```diff
@@ -49,6 +49,9 @@ CIRQUE_TESTS=(
     "CommissioningFailureOnReportTest"
     "PythonCommissioningTest"
     "CommissioningWindowTest"
+    "SubscriptionResumptionTest"
```
Comment (andy31415) [inline scripts/tests/cirque_tests.sh:52]: Thse seem to fail .... are these supposed to be added?

**Comment by andy31415** [comment]
@Jerry-ESP please create a followup PR to fix the cirque tests. If they cannot be fixed, I will have to roll back the PR to keep the 1.2 CI green.

File: scripts/tests/cirque_tests.sh (hunk 1 above)
Comment (Jerry-ESP) [reply scripts/tests/cirque_tests.sh:52]: OK, I will create another PR to fix this issue

**Comment by Jerry-ESP** [comment]
> @Jerry-ESP please create a followup PR to fix the cirque tests. If they cannot be fixed, I will have to roll back the PR to keep the 1.2 CI green.

@andy31415  This PR: https://github.com/project-chip/connectedhomeip/pull/33207 can fix this error

**Pull request merged**